src/components/reusable/daterangepicker/daterangepicker.ts

```diff
@@ -0,0 +1,287 @@
+        for=${this.name}
+        ?disabled=${this.disabled}
+      >
+        ${this.required ? html`<span class="required">*</span>` : null}
+        <slot></slot>
+      </label>
+
+      <div class="input-wrapper">
+        <input
+          class="${classMap({
+            'size--sm': this.size === 'sm',
+            'size--lg': this.size === 'lg',
+          })}"
+          type="date"
+          id=${this.name ? `${this.name}-1` : 'date-range-1'}
+          name=${this.name}
+          value=${this.startDate}
+          ?required=${this.required}
+          ?disabled=${this.disabled}
+          ?invalid=${this.invalidText !== ''}
+          min=${ifDefined(this.minDate)}
+          max=${ifDefined(this.maxDate)}
+          step=${ifDefined(this.step)}
+          @input=${(e: any) => this.handleStartDate(e)}
+        />
+      </div>
+
+      <span class="range-span">—</span>
+      <div class="input-wrapper">
+        <input
+          class="${classMap({
+            'size--sm': this.size === 'sm',
+            'size--lg': this.size === 'lg',
+          })}"
+          type="date"
+          id=${this.name ? `${this.name}-2` : 'date-range-2'}
+          value=${this.endDate}
+          ?disabled=${this.disabled}
+          ?invalid=${this.invalidText !== ''}
+          min=${ifDefined(this.startDate ?? '')}
+          max=${ifDefined(this.maxDate)}
+          step=${ifDefined(this.step)}
+          @input=${(e: any) => this.handleEndDate(e)}
+        />
+      </div>
+
+      ${this.caption !== ''
+        ? html` <div class="caption">${this.caption}</div> `
+        : null}
+      ${this.invalidText !== ''
+        ? html` <div class="error">${this.invalidText}</div> `
+        : null}
+      ${this.warnText !== '' && this.invalidText === ''
+        ? html`<div class="warn">${this.warnText}</div>`
+        : null}
+    `;
+  }
+
+  override updated(changedProps: PropertyValues) {
+    if (changedProps.has('startDate')) {
+      // set form data value
+      this.internals.setFormValue('start-date', this.startDate);
+      this.internals.setValidity({});
+      this.invalidText = '';
+      // set validity
+      if (this.required && (!this.startDate || this.startDate === '')) {
+        this.internals.setValidity(
+          { valueMissing: true },
+          'Both dates are required.'
+        );
+        this.invalidText = this.internals.validationMessage;
+        return;
+      }
+      // validate min
+      if (this.startDate !== '' && this.minDate !== '') {
+        this.validateMinDate(this.startDate);
+      }
+      //validate max
+      if (this.startDate !== '' && this.maxDate !== '') {
+        this.validateMaxDate(this.startDate);
+      }
+      // validate start & end date
+      if (this.startDate !== '' && this.endDate !== '') {
+        this.validateStartEndDate();
+      }
+    }
+    if (changedProps.has('endDate')) {
+      this.internals.setFormValue('end-date', this.endDate);
```

Comment:
Use `this.name + '-end'` here instead of 'end-date'.

Reply:
done